tests/imap/MailCollector.php

```diff
@@ -0,0 +1,303 @@
+            'mail_server'  => '127.0.0.1',
+            'server_type'  => '/imap',
+            'server_port'  => 143,
+            'server_ssl'   => '',
+            'server_cert'  => '/novalidate-cert'
+         ])
+      )->isGreaterThan(0);
+
+      $this->boolean($collector->getFromDB($collector->fields['id']))->isTrue();
+      $this->string($collector->fields['host'])->isIdenticalTo('{127.0.0.1:143/imap}');
+      $collector->connect();
+      $this->variable($collector->fields['errors'])->isEqualTo(0);
+   }
+
+   public function testCollect() {
+      global $DB;
+
+      //assign email to user
+      $nuid = getItemByTypeName('User', 'normal', true);
+      $uemail = new \UserEmail();
+      $this->integer(
+         (int)$uemail->add([
+            'users_id'     => $nuid,
+            'is_default'   => 1,
+            'email'        => 'normal@glpi-project.org'
+         ])
+      )->isGreaterThan(0);
+      $tuid = getItemByTypeName('User', 'tech', true);
+      $this->integer(
+         (int)$uemail->add([
+            'users_id'     => $tuid,
+            'is_default'   => 1,
+            'email'        => 'tech@glpi-project.org'
+         ])
+      )->isGreaterThan(0);
+
+      $ticket = new \Ticket();
+      $tid = $ticket->add([
+         'name'                  => 'database issue',
+         'content'               => 'It seems one field from the last migration has not been added in my database.',
+         '_users_id_requester'   => $tuid
+      ]);
+      $this->integer($tid)->isGreaterThan(0);
+
+      $collector = new \mock\MailCollector;
+      $this->calling($collector)->getMessageMatch = "/GLPI-(1155)\.[0-9]+\.[0-9]+@\w*/";
+      $this->collector = $collector;
+
+      $this->doConnect();
+      $msg = $collector->collect();
+      $this->variable($msg)->isIdenticalTo('Number of messages: available=5, retrieved=5, refused=1, errors=0, blacklisted=0');
+      $rejecteds = iterator_to_array($DB->request(['FROM' => \NotImportedEmail::getTable()]));
+
+      $this->array($rejecteds)->hasSize(1);
+      $this->array(array_pop($rejecteds))
+         ->variable['from']->isIdenticalTo('unknwon@glpi-project.org')
+         ->variable['reason']->isEqualTo(\NotImportedEmail::USER_UNKNOWN);
+
+      $iterator = $DB->request([
+         'SELECT' => ['t.id', 't.name', 'tu.users_id'],
+         'FROM'   => \Ticket::getTable() . " AS t",
+         'INNER JOIN'   => [
+            \Ticket_User::getTable() . " AS tu"  => [
+               'ON'  => [
+                  't'   => 'id',
+                  'tu'  => 'tickets_id'
+               ]
+            ]
+         ],
+         'WHERE'  => [
+            'tu.users_id'  => $tuid,
+            'tu.type'      => \CommonITILActor::REQUESTER
+         ]
+      ]);
+
+      $this->integer(count($iterator))->isIdenticalTo(3);
+      $names = [];
+      while ($data = $iterator->next()) {
+         $names[] = $data['name'];
+      }
+
+      $expected_names = [
+         'database issue',
```

Comment:
In eml files, there is only 2 files having `From: Tech Ni Cian <tech@glpi-project.org>`. Where does this one comes from ?

Reply:
Created from unit tests, to try to add a followup from collector